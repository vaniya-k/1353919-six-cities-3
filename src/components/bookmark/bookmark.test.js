import React from 'react';
import renderer from 'react-test-renderer';
import Bookmark from './bookmark.jsx';
import {createStore} from "redux";
import {Provider} from "react-redux";

const mockReducer = () => {
  const state = {
    offers: null,
    user: {
      authorizationStatus: `AUTH`
    }
  };
  return state;
};

const mockStore = createStore(mockReducer);

it(`A regular bookmark for a unsaved place`, () => {
  const tree = renderer
    .create(<Provider store={mockStore}><Bookmark isFavorite={false} id={1} isOnPlacePage={false}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`A regular bookmark for a saved place`, () => {
  const tree = renderer
    .create(<Provider store={mockStore}><Bookmark isFavorite={true} id={1} isOnPlacePage={false}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`A big bookmark for an unsaved place`, () => {
  const tree = renderer
    .create(<Provider store={mockStore}><Bookmark isFavorite={false} id={1} isOnPlacePage={true}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`A big bookmark for an saved place`, () => {
  const tree = renderer
    .create(<Provider store={mockStore}><Bookmark isFavorite={true} id={1} isOnPlacePage={true}/></Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
