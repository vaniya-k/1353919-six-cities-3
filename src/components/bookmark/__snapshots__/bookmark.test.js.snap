// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A big bookmark for an saved place 1`] = `
<button
  className="property__bookmark-button property__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      href="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`A big bookmark for an unsaved place 1`] = `
<button
  className="property__bookmark-button button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="property__bookmark-icon"
    height="33"
    width="31"
  >
    <use
      href="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`A regular bookmark for a saved place 1`] = `
<button
  className="place-card__bookmark-button place-card__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      href="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;

exports[`A regular bookmark for a unsaved place 1`] = `
<button
  className="place-card__bookmark-button button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="19"
    width="18"
  >
    <use
      href="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;
